File: 566_ReshapeTheMatrix.js
/**
 * @param {number[][]} nums
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(nums, r, c) {
  if (nums.length === 0 || nums.length * nums[0].length !== r * c) {
    return nums;
  }

  var i1 = 0;
  var j1 = -1;
  var result = [];
  for (var i2 = 0; i2 < r; i2 += 1) {
    var tmpArr = [];
    for (var j2 = 0; j2 < c; j2 += 1) {
      if (j1 >= nums[0].length - 1) {
        i1 += 1;
        j1 = 0;
      } else {
        j1 += 1;
      }
      tmpArr.push(nums[i1][j1]);
    }
    result.push(tmpArr);
    tmpArr = [];
  }

  return result;
};

File: 566.ReshapetheMatrix.java
class Solution {
    public int[][] matrixReshape(int[][] nums, int r, int c) {
        int r1 = nums.length;
        int c1 = nums[0].length;
        int numsCount = r1 * c1;
        if (numsCount != r * c) {
            return nums;
        }
        int[][] result = new int[r][c];
        for (int i = 0; i < numsCount; i += 1) {
            result[i / c][i % c] = nums[i / c1][i % c1];
        }
        return result;
    }
}